include/RAJA/policy/atomic_builtin.hpp

```diff
@@ -37,389 +44,828 @@ namespace RAJA
 struct builtin_atomic {
 };
 
-namespace detail
-{
 
-template <std::size_t BYTES>
-struct BuiltinAtomicTypeImpl {
-  static_assert(!(BYTES == sizeof(unsigned) ||
-                  BYTES == sizeof(unsigned long long)),
-                "Builtin atomic operations require targets that match the size of 'unsigned int' or 'unsigned long long' (usually 4 or 8 bytes).");
+namespace detail {
+
+
+/*!
+ * Type trait for determining if the operator should ALWAYS be implemented
+ * using a compare and swap loop
+ */
+template <typename T>
+struct builtin_alwaysUseCAS {
```

Comment:
The difference between this and builtin_useCAS confuses me.

Reply:
I removed builtin_alwaysUseCAS. It was for atomics that had no intrinsic support, so the only way for them to be implemented was through a compare and swap loop.

Reply:
But you're right that it was not intuitive.